### PR TITLE
docs(serialization): Update for IApplicationBuilder

### PR DESCRIPTION
**Pull request by lukeblevins**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/Overview/Serialization/SerializationOverview.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ uid: Overview.Serialization
 
 # Serialization
 
-Accessing the serialized and deserialized representation of an object can be important for dynamic, data-rich applications. `Uno.Extensions.Serialization` allows for simplified access to serializer objects as dependencies. This library supports the new serialization [technique](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator) powered by code generation.
+**Serialization** involves converting an object into a format that can be easily stored or transmitted. On the receiving end, the object is reconstructed back into its original form through **deserialization**. These two operations complement each other and can be important for dynamic, data-rich applications. `Uno.Extensions.Serialization` allows for simplified access to serializer objects as dependencies. This library supports the new serialization [technique](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator) powered by code generation.
```
Comment (weitzhandler) [inline doc/Overview/Serialization/SerializationOverview.md:7]:
```suggestion
**Serialization** involves converting an object into a format that can be easily stored or transmitted. On the receiving end, the object is reconstructed back into its original form through **deserialization**. These two operations complement each other and can be important for dynamic, data-rich applications.  
`Uno.Extensions.Serialization` allows for simplified access to serializer objects as dependencies. This library supports the new serialization [technique](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator) powered by code generation.
```

File: doc/Overview/Serialization/SerializationOverview.md [hunk 2]
```diff
@@ -1,43 +1,88 @@
 ---
 uid: Overview.Serialization
 ---
-## Serialization
 
-Register serializer that implements `ISerializer`
+# Serialization
+
+**Serialization** involves converting an object into a format that can be easily stored or transmitted. On the receiving end, the object is reconstructed back into its original form through **deserialization**. These two operations complement each other and can be important for dynamic, data-rich applications. `Uno.Extensions.Serialization` allows for simplified access to serializer objects as dependencies. This library supports the new serialization [technique](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator) powered by code generation.
```
Comment (weitzhandler) [inline doc/Overview/Serialization/SerializationOverview.md:7]: Add linebreaks when paragraphs become too large. It's more readable.
Adding a linebreak in markdown is achieved by adding a double space at the end of the line.

```suggestion
**Serialization** involves converting an object into a format that can be easily stored or transmitted. On the receiving end, the object is reconstructed back into its original form through **deserialization**. These two operations complement each other and can be important for dynamic, data-rich applications.  
`Uno.Extensions.Serialization` allows for simplified access to serializer objects as dependencies. This library supports the new serialization [technique](https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-source-generator) powered by code generation.
```

Comment (weitzhandler) [reply doc/Overview/Serialization/SerializationOverview.md]: @lukeblevins you don't always have to add an empty line in between, it sometimes makes sense to just wrap the current line.

File: doc/Overview/Serialization/SerializationOverview.md [hunk 3]
```diff
@@ -1,43 +1,133 @@
...
+        });
+...
+```
+
+#### Source Generation
+
+As of .NET 6, a code generation-enabled serializer is supported. The type to serialize is named `Person` in this example.
+
+```csharp
+public class Person
```
Comment (nickrandolph) [inline doc/Overview/Serialization/SerializationOverview.md:72]: Can we change this to a record?

File: doc/Overview/Serialization/SerializationOverview.md [hunk 4]
```diff
@@ -1,43 +1,133 @@
...
-    // ........ //
 }
 ```
+
+The `PersonJsonContext` class is then used to serialize and deserialize instances of `Person`:
+
+```csharp
+var person = new Person("Megan", 23, 1.8, 90.0);
+
+var json = JsonSerializer.Serialize(person!, PersonJsonContext.Default.Person);
```
Comment (nickrandolph) [inline doc/Overview/Serialization/SerializationOverview.md:107]: The whole point is to use ISerializer as part of Serizalization. You need to register the PersonJsonContext.Default as part of app builder. See https://github.com/unoplatform/uno.extensions/blob/18f534042d9204ea3ad5962fbc517903c953902b/samples/Playground/Playground.Shared/AppHost.cs#L35-L38

**Review by nickrandolph — APPROVED** [review]

**Pull request merged**